presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -301,6 +334,24 @@ private synchronized void refreshNodesInternal()
         // nodes by connector id changes anytime a node adds or removes a connector (note: this is not part of the listener system)
         activeNodesByConnectorId = byConnectorIdBuilder.build();
 
+        SetView<String> deadNodeIds = difference(
```

Comment:
Perhaps, it makes more sense to record both `allNodes` and `activeNodes` here. Since this is the source of all changes, we might want the resilient config to guard here, so that if the config is disabled, the `allNodes` is the same as `activeNodes`. This kill switch will be closer to the source so in case the logic here is flawed, we could turn it off completely. 

And by doing this, we don't need the other resilient check as well, because if this is turned off, they will just behave as before, where only active nodes are returned, regardless of whether it wants allNodes or activeNodes logically.

Reply:
Yeah makes sense. Added new config parameter `internal-communication.memoize-dead-nodes-enabled` and made changes to have this code block run only when the config is set to true.